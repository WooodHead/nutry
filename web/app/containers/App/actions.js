import {
  SEARCH_COMPLETE,
} from 'containers/Search/constants';

import {
  DEFAULT_ACTION,
  LOGIN_CLICKED,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function login() {
  return {
    type: LOGIN_CLICKED,
  };
}
export function loginSuccess(token) {
  return {
    type: LOGIN_SUCCESS,
    token,
  };
}

export function searchComplete(searchResults) {
  return {
    type: SEARCH_COMPLETE,
    searchResults,
  };
}

export function loginFailure(err) {
  return {
    type: LOGIN_FAILURE,
    err,
  };
}
