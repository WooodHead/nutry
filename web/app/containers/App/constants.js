/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const DEFAULT_LOCALE = 'en';
/*
 *
 * LandingPage constants
 *
 */
const prefix = (name) => `app/App/${name}`;
export const DEFAULT_ACTION = prefix('DEFAULT_ACTION');
export const LOGIN_FAILURE = prefix('LOGIN_FAILURE');
export const LOGIN_SUCCESS = prefix('LOGIN_SUCCESS');
export const LOGIN_CLICKED = prefix('LOGIN_CLICKED');
