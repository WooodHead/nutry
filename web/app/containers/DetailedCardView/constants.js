/*
 *
 * DetailedCardView constants
 *
 */

export const DEFAULT_ACTION = 'app/DetailedCardView/DEFAULT_ACTION';
