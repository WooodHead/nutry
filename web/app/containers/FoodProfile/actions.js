/*
 *
 * FoodProfile actions
 *
 */
import { defaultPortions } from 'lib/nutrientMap';
import {
  DEFAULT_ACTION,
  ON_BACK,
  GET_PROFILE,
  TAB_CHANGED,
  PORTION_CHANGED,
  AGE_GROUP_CHANGED,
  NUTRIENT_SELECTED,
  GET_PROFILE_FAILURE,
  GET_PROFILE_SUCCESS,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function loadProfile(serialNumber, source) {
  return {
    type: GET_PROFILE,
    serialNumber,
    source,
  };
}
export function loadProfileSuccess(profileBody) {
  return {
    type: GET_PROFILE_SUCCESS,
    nutrientsById: profileBody.nutrients,
    portionsAvailable: defaultPortions(profileBody.portions),
  };
}
export function loadProfileFailure(error) {
  return {
    type: GET_PROFILE_FAILURE,
    error,
  };
}
export function tabChanged(tab) {
  return {
    type: TAB_CHANGED,
    tab,
  };
}
export function nutrientSelected(prefix, id) {
  return {
    type: NUTRIENT_SELECTED,
    prefix,
    id,
  };
}
export function portionChanged(portionSelected) {
  return {
    type: PORTION_CHANGED,
    portionSelected,
  };
}
export function ageGroupChanged(ageGroupSelected) {
  return {
    type: AGE_GROUP_CHANGED,
    ageGroupSelected,
  };
}

export function onBack() {
  return {
    type: ON_BACK,
  };
}
export function clearFoodProfile() {
  return {
    type: ON_BACK,
  };
}
