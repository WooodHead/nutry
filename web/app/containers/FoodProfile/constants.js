/*
 *
 * FoodProfile constants
 *
 */
import { Map, List } from 'immutable';
const prefix = (name) => `app/FoodProfile/${name}`;

export const ON_BACK = prefix('ON_BACK');

export const GET_PROFILE = prefix('GET_PROFILE');
export const GET_PROFILE_SUCCESS = prefix('GET_PROFILE_SUCCESS');
export const GET_PROFILE_FAILURE = prefix('GET_PROFILE_FAILURE');

export const TAB_CHANGED = prefix('TAB_CHANGED');
export const PORTION_CHANGED = prefix('PORTION_CHANGED');
export const NUTRIENT_SELECTED = prefix('NUTRIENT_SELECTED');
export const AGE_GROUP_CHANGED = prefix('AGE_GROUP_CHANGED');

export const HEADER = 'HEADER';
export const PARENT_ROW = 'PARENTR_ROW';
export const PARENT_NONAME_ROW = 'PARENT_NONAME_ROW';
export const CHILD_ROW = 'CHILD_ROW';
export const CHILD2_ROW = 'CHILD2_ROW';
export const SUMMARY_IDS = [
  {
    prefix: 'ENERC',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'ENERC_KCAL',
    type: PARENT_NONAME_ROW,
    selected: false,
  },
  {
    prefix: 'CHOCDF',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'SUGAR',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'FIBTG',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'PROCNT',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'FAT',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'NA',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'K',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'VITC',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'VITD',
    type: PARENT_ROW,
    selected: false,
  },
  {
    prefix: 'WATER',
    type: PARENT_ROW,
    selected: false,
  },
];
export const DETAILED_IDS = [
  // energy
  'ENERC', 'ENERC_KCAL',
  // fats
  'FAT', 'FASAT', 'FAMS', 'FAPU', 'FATRN', 'LCW3TOTAL',
  // carbohydrates
  'CHOCDF', 'FIBTG', 'SUGAR', 'GLUS', 'MALS', 'FRUS', 'LACS', 'GALS',
  // proteins
  'PROCNT', 'TRP', 'THR', 'ILE', 'LEU', 'LYS', 'MET', 'CYS', 'PHE', 'TYR', 'VAL',
  'ARG', 'HISTN', 'ALA', 'ASP', 'GLU', 'PRO', 'SER', 'HYP',
  // minerals
  'CA', 'FE', 'MG', 'P', 'K', 'NA', 'ZN', 'CU', 'MN', 'FLD',
  // vitamins
  'VITC', 'VITA_IU', 'RETOL', 'VITA_RAE', 'CARTA', 'CARTB', 'CRYPX', 'LYCPN',
  'LUT+ZEA', 'VITD', 'TOCPHA', 'TOCPHB', 'TOCPHD', 'VITK', 'THIA', 'NIA', 'SE',
  'FOL', 'FOLFD', 'FOLAC', 'FOLDFE', 'VITB6A', 'CHOLN', 'BETN',
  // basics
  'WATER',
];
export const DETAILED_SECTIONS = {
  energy: {
    headerName: 'Energy',
    items: [
      {
        prefix: 'ENERC',
        type: PARENT_ROW,
        selected: false,
      }, {
        prefix: 'ENERC_KCAL',
        type: PARENT_NONAME_ROW,
        selected: false,
      }],
  },
  fat: {
    headerName: 'Fat & Fatty Acids',
    items: [
      {
        prefix: 'FAT',
        type: PARENT_ROW,
        selected: false,
      }, {
        prefix: 'FASAT',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FAMS',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FAPU',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'FATRN',
        type: CHILD_ROW,
        selected: false,
      }, {
        prefix: 'LCW3TOTAL',
        type: PARENT_ROW,
        selected: false,
      }],
  },
  carbohydrates: {
    headerName: 'Carbohydrates',
    items: [
      {
        prefix: 'CHOCDF',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FIBTG',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'SUGAR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'GLUS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'SUCS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'MALS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'FRUS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'LACS',
        type: CHILD2_ROW,
        selected: false,
      },
      {
        prefix: 'GALS',
        type: CHILD2_ROW,
        selected: false,
      },
    ],
  },
  protein: {
    headerName: 'Proteins',
    items: [
      {
        prefix: 'PROCNT',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TRP',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'THR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ILE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LEU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LYS',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'MET',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CYS',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'PHE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'TYR',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VAL',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ARG',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'HISTN',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ALA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'ASP',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'GLU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'PRO',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'SER',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'HYP',
        type: CHILD_ROW,
        selected: false,
      },
    ],
  },
  minerals: {
    headerName: 'Minerals',
    items: [
      {
        prefix: 'CA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'MG',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'P',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'K',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'NA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'ZN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'CU',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'MN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'SE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FLD',
        type: PARENT_ROW,
        selected: false,
      },
    ],
  },
  vitamins: {
    headerName: 'Vitamins',
    items: [
      {
        prefix: 'VITC',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'VITA_IU',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'RETOL',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITA_RAE',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CARTA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CARTB',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'CRYPX',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LYCPN',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'LUT+ZEA',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITD',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHB',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'TOCPHD',
        type: CHILD_ROW,
        selected: false,
      },
      {
        prefix: 'VITK',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'THIA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'NIA',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'SE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOL',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLFD',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLAC',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'FOLDFE',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'VITB6A',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'CHOLN',
        type: PARENT_ROW,
        selected: false,
      },
      {
        prefix: 'BETN',
        type: PARENT_ROW,
        selected: false,
      },
    ],
  },
};
export const FILTERS = {
  summary: [
    'CHOCDF',
    'SUGAR',
    'FIBTG',
    'PROCNT',
    'FAT',
    'WATER',
  ],
  detailed: [
    'CHOCDF',
    'SUGAR',
    'FIBTG',
    'PROCNT',
    'FAT',
    'WATER',
  ],
};
export const INITIAL_STATE = {
  loading: true,
  serialNumber: false,
  profileHeader: {},
  error: false,
  source: false,
  nutrients: {
    byId: {},
    bySection: Map(DETAILED_SECTIONS),
    bySummaryIds: SUMMARY_IDS,
    byDetailedIds: DETAILED_IDS,
    bySummaryPie: List([]),
    byDetailedPie: List([]),
  },
  tabSelected: 'summary',
  idSelected: false,
  nutrientSelected: '',
  portionSelected: { amt: 1,
    className: 'per100g',
    g: 100,
    label: 'per 100g',
    unit: 'per 100g',
    value: 100,
  },
  portionsAvailable: [],
  ageGroupSelected: { value: 'AM19', label: 'Adult Male (19-30)', className: 'am-19' },
};
