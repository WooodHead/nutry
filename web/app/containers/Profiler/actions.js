/*
 *
 * Profiler actions
 *
 */

import {
  CHANGE_PROFILER_SEARCH,
  SEARCH_COMPLETE,
  SEARCH_FAILURE,
  SEARCH_LOADING,
  ADD_PROFILER_ELEMENT,
  DELETE_PROFILER_ELEMENT,
  UPDATE_PROFILER_ELEMENT,
} from './constants';

export function changeSearch() {
  return {
    type: CHANGE_PROFILER_SEARCH,
  };
}
export function searchComplete(results) {
  if (results) {
    return {
      type: SEARCH_COMPLETE,
      results,
    };
  }
  return {
    type: SEARCH_COMPLETE,
    results: [],
  };
}
export function searchLoading() {
  return {
    type: SEARCH_LOADING,
  };
}
export function searchFailure(error) {
  return {
    type: SEARCH_FAILURE,
    error,
  };
}
export function addProfilerElement() {
  return {
    type: ADD_PROFILER_ELEMENT,
  };
}
export function deleteProfilerElement(id) {
  return {
    type: DELETE_PROFILER_ELEMENT,
    id,
  };
}
export function updateProfilerElement(id, key, value) {
  return {
    type: UPDATE_PROFILER_ELEMENT,
    id,
    key,
    value,
  };
}
