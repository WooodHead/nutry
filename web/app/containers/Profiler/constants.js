/*
 *
 * Profiler constants
 *
 */
const prefix = (name) => `app/Profiler/${name}`;

export const CHANGE_PROFILER_SEARCH = prefix('CHANGE_PROFILER_SEARCH');
export const SEARCH_COMPLETE = prefix('SEARCH_COMPLETE');
export const SEARCH_LOADING = prefix('SEARCH_LOADING');
export const SEARCH_FAILURE = prefix('SEARCH_FAILURE');
export const ADD_PROFILER_ELEMENT = prefix('ADD_PROFILER_ELEMENT');
export const DELETE_PROFILER_ELEMENT = prefix('DELETE_PROFILER_ELEMENT');

export const UPDATE_PROFILER_ELEMENT = prefix('UPDATE_PROFILER_ELEMENT');
