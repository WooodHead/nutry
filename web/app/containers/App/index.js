/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import { compose } from 'redux';
import Search from 'containers/Search/Loadable';
import FoodProfile from 'containers/FoodProfile/Loadable';
import LandingPage from 'containers/LandingPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import reducer from './reducer';
import saga from './sagas';

export class App extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <div>
        <Switch>
          <Route exact path="/" component={Search} />
          <Route path="/foodprofile/:profileId" component={FoodProfile} />
          <Route exact path="/overview" component={LandingPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    );
  }
}
const withReducer = injectReducer({ key: 'global', reducer });
const withSaga = injectSaga({ key: 'global', saga });

export default compose(
  withReducer,
  withSaga,
)(App);
