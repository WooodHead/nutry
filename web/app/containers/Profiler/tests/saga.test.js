// import { expectSaga } from 'redux-saga-test-plan';
import { takeLatest, put } from 'redux-saga/effects';
import { searchComplete } from 'containers/App/actions';
import { elements } from 'fixtures/profiler';
import { getProfilerParsedSuccess } from 'mocks/getElasticsearchQueryMock';
import { CHANGE_SEARCH } from 'containers/Search/constants';
import { CHANGE_PROFILER_SEARCH } from '../constants';
import { searchFailure } from '../actions';

import searcProfilerData, { getResults } from '../saga';


describe('Container [Profiler] - sagas', () => { /* eslint-disable redux-saga/yield-effects */
  describe('getResults Saga', () => {
    let getResultsGenerator;
    beforeEach(() => {
      getResultsGenerator = getResults();
      // debounce delay
      const callDelayDescriptor = getResultsGenerator.next().value;
      expect(callDelayDescriptor).toMatchSnapshot();
      // select searchString value
      const selectSearchStringDescriptor = getResultsGenerator.next('chocolate').value;
      expect(selectSearchStringDescriptor).toMatchSnapshot();
      // select element list
      const selectElementDescriptor = getResultsGenerator.next(elements).value;
      expect(selectElementDescriptor).toMatchSnapshot();
      // select results of search
      // call axios getFoodProfile
      const callDescriptor = getResultsGenerator.next(getProfilerParsedSuccess).value;
      expect(callDescriptor).toMatchSnapshot();
    });
    it('should dispatch searchComplete action result fetched successfully', () => {
      const response = getProfilerParsedSuccess;
      const putDescriptor = getResultsGenerator.next(response).value;
      expect(putDescriptor).toEqual(put(searchComplete(response)));
    });
    it('should dispatch searchFailure after failure to fetch', () => {
      const response = new Error('Fetch Failure');
      const descriptor = getResultsGenerator.throw(response).value;
      expect(descriptor).toMatchSnapshot();
      expect(descriptor).toEqual(put(searchFailure(response)));
    });
  });
  describe('searcProfilerData Saga', () => {
    const searcProfilerSaga = searcProfilerData();
    it('works with unit tests', () => {
      const takeLatestDescriptor = searcProfilerSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest([CHANGE_SEARCH, CHANGE_PROFILER_SEARCH], getResults));
    });
  });
});
