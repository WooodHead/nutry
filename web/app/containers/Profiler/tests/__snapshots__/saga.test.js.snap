// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchComplete action result fetched successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      400,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchComplete action result fetched successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchComplete action result fetched successfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchComplete action result fetched successfully 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      Object {
        "uuid1": Object {
          "nutrient": Object {
            "className": "elements-water",
            "label": "Water",
            "value": "WATER",
          },
          "scale": 50,
        },
      },
      Object {
        "hits": 1,
        "items": Array [
          Object {
            "_id": "AWF8pHZ9bw9zRJtz161q",
            "_index": "nutry-names",
            "_score": 10,
            "_source": Object {
              "SN": "02049",
              "alias": Array [],
              "allergen": Array [],
              "group": "Spices and Herbs",
              "name": "Fresh thyme",
              "nutrients": Object {},
              "portions": Array [],
              "tags": Array [],
              "usage": Array [],
            },
          },
        ],
        "max_score": 10,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchFailure after failure to fetch 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      400,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchFailure after failure to fetch 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchFailure after failure to fetch 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchFailure after failure to fetch 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      Object {
        "uuid1": Object {
          "nutrient": Object {
            "className": "elements-water",
            "label": "Water",
            "value": "WATER",
          },
          "scale": 50,
        },
      },
      Object {
        "hits": 1,
        "items": Array [
          Object {
            "_id": "AWF8pHZ9bw9zRJtz161q",
            "_index": "nutry-names",
            "_score": 10,
            "_source": Object {
              "SN": "02049",
              "alias": Array [],
              "allergen": Array [],
              "group": "Spices and Herbs",
              "name": "Fresh thyme",
              "nutrients": Object {},
              "portions": Array [],
              "tags": Array [],
              "usage": Array [],
            },
          },
        ],
        "max_score": 10,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [Profiler] - sagas getResults Saga should dispatch searchFailure after failure to fetch 5`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "error": [Error: Fetch Failure],
      "type": "app/Profiler/SEARCH_FAILURE",
    },
    "channel": null,
  },
}
`;
